opflexagent/gbp_ovs_agent.py

```diff
@@ -277,15 +277,70 @@ def port_bound(self, port, net_uuid,
                     port, net_uuid, network_type, physical_network,
                     segmentation_id, fixed_ips, device_owner, ovs_restarted)
         elif network_type in self.supported_pt_network_types:
+            # PT cleanup is needed before writing the new endpoint files
+            self.mapping_cleanup(port.vif_id, cleanup_vrf=False)
             if ((self.opflex_networks is None) or
                     (physical_network in self.opflex_networks)):
                 # Port has to be untagged due to a opflex agent requirement
                 self.int_br.clear_db_attribute("Port", port.port_name, "tag")
-                self.mapping_to_file(port, net_uuid, mapping, fixed_ips,
+                # Multiple files will be created based on how many MAC
+                # addresses are owned by the specific port.
+                mapping_copy = copy.deepcopy(mapping)
+                mac_active_aap = {}
+                mapping_copy['allowed_address_pairs'] = []
+                mapping_copy['floating_ip'] = []
+                fip_by_fixed = {}
+                original_mac = mapping.get('mac_address') or port.vif_mac
+                # Prepare FIPs by fixed_ip
+                for fip in mapping.get('floating_ip', []):
+                    fip_by_fixed.setdefault(
+                        fip['fixed_ip_address'], []).append(fip)
+                # For the main MAC, set floating IP collection to all those
+                # FIPs pointing to the active Port fixed ips.
+                for fixed in fixed_ips:
+                    mapping_copy['floating_ip'].extend(
+                        fip_by_fixed.get(fixed['ip_address'], []))
+
+                # For the main MAC EP, set al the AAP with no mac address or
+                # MAC address equal to the original MAC.
+                for aap in mapping.get('allowed_address_pairs', []):
+                    if aap.get('active'):
+                        if not aap.get('mac_address') or aap.get(
+                                'mac_address') == original_mac:
+                            # Should go with the MAIN mac address EP file
+                            mapping_copy['allowed_address_pairs'].append(aap)
+                            # Also set the right floating IPs
+                            mapping_copy['floating_ip'].extend(
+                                fip_by_fixed.get(aap['ip_address'], []))
```

Comment:
Not sure what you mean, this gets the right floating IP based on the AAPs fixed ips.